### ISSUE TITLE
allow returning a value on `else`

### ISSUE THREAD

**Issue by yonilevy** [alexschrod/ward#1]
Hi,

Nice crate, just what I was looking for :)

One thing that'd make it more like Swift's guard would be to allow returning values in the `else` portion, e.g.:

```rust
fn foo(x: Option<i32>) -> i32 {
    guard!(let x = x, else { 42 });
    // Do more stuff with x...
}
```

WDYT?

Thanks!

**Comment by ilyvion** [comment]
That sounds like it belongs in there. I'll see what I can do. :)

**Comment by ilyvion** [comment]
I kind of regret coding it with auto-return initially, because removing that counts as a major semver version increase. Not that that's the end of the world, but anyway: This is now released as v2.0.

**Comment by yonilevy** [comment]
Awesome! Thank you :)
